packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -165,13 +165,10 @@ class FlutterManifest {
   ///   licenses:
   ///     - assets/foo_license.txt
   /// ```
-  List<String> get additionalLicenses {
-    final Object? licenses = _flutterDescriptor['licenses'];
-    if (licenses is YamlList) {
-      return licenses.map((Object? element) => element.toString()).toList();
-    }
-    return <String>[];
-  }
+  List<String> get additionalLicenses => <String>[
+    if (_flutterDescriptor case {'licenses': final YamlList list})
+      for (final dynamic item in list) '$item',
```

Comment:
We should definitely definitely avoid use of `dynamic`.
https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-var-and-dynamic

Reply:
I know there are plenty of exceptions visible in this diff alone, I point out this one because before this refactor, this code section had no dynamic.